src/webSession.js

```diff
@@ -142,16 +143,30 @@ WebSession.prototype.handleURL = async function () {
 		translate.setCookieSandbox(this._cookieSandbox);
 		
 		try {
-			await HTTP.processDocuments(
-				[url],
-				(doc) => {
-					translate.setDocument(doc);
-					// This could be optimized by only running detect on secondary translators
-					// if the first fails, but for now just run detect on all
-					return translate.getTranslators(true);
-				},
-				this._cookieSandbox
+			let req = await Zotero.HTTP.request(
+				"GET",
+				url,
+				{
+					responseType: 'document',
+					fallbackToPDF: true,
+					cookieSandbox: this._cookieSandbox
+				}
 			);
+			if(req.type === 'document') {
+				translate.setDocument(req.response);
+				// This could be optimized by only running detect on secondary translators
+				// if the first fails, but for now just run detect on all
+				translate.getTranslators(true);
+			}
+			// If PDF received
+			else {
+				let uploadID = await Recognizer.upload(req.response);
+				let item = await Recognizer.recognize(uploadID);
+				this.ctx.response.body = Zotero.Utilities.itemToAPIJSON(item);
+				await Recognizer.remove(uploadID);
```

Comment:
Seems like here and in `handleProcess` we should do the `remove` in a `try`/`catch` within a `finally` so that the file is removed even if something fails. (We'll have a lifecycle policy on the bucket, but we should still try to clean up properly.)